widget/embedded/src/utils/common.ts

```diff
@@ -41,6 +41,22 @@ export function debounce(fn: Function, time: number) {
   }
 }
 
+export function createCustomEvent<T extends Record<string, unknown>>(
```

Comment:
I suggest this function to something like this:

```js
function emitPreventableEvent(type, payload, action){
  let defaultPrevented = false;
  const extendedPayload = {
    preventDefault(){ ... }
    ...payload, 
  };
  
  // `mitt` will run syncronously after emit an event. so we are sure an event can be prevented before running action.
  eventEmitter.emit(type, extendedPayload);
  
  if(!defaultPrevented){
    action()
  }
}
```

Reply:
In this way, you don't need to put a reference in useRef. You can use this util whenever you want to emit the event.

Reply:
Also i suggest to move this util to `/utils/event.ts`

Reply:
I agree with these changes to avoid repetitive checks when using this function. However, to achieve this functionality, you need to store the result of this function somewhere for a specific event to preserve its closure.